WebRoot/WEB-INF/tags/infra/skinAndRedirect.tag

```diff
@@ -28,6 +29,10 @@
 		PageContext pageContext = (PageContext)jspContext;
 		String skin = (String)pageContext.findAttribute("skin");
 
+		if (!Pattern.matches("^[0-9A-Za-z]+$", skin)) {
```

Comment:
any reason to not use already available zm:cook method https://github.com/Zimbra/zm-web-client/blob/09e633765546281efdb3d6ee84f2fe82771064c0/WebRoot/public/launchNewWindow.jsp#L142 ?
and having custom implementation

Reply:
@silentsakky Thanks for reference to cook method! However, the goal of this commit is not only to eradicate directory traversal bug, but also not to trigger additional errors DoS'ing the logs. If a skin method is escaped only, it becomes "%2F..%2F..%2Fetc%2Flocalconfig.xml" throwing the exception of BundleNotFound. My implementation enforces the business logic rule that skin name is one single word, thus narrowing the manipulation space to a non-existent skin name. But non-existent skin name is handled already, producing no additional log entries. So custom implementation here provides closer approximation to the expected logic.

Reply:
well I want logs to be added when someone passes some weird data and tries to do xss attack, that would be helpful for admins instead of silently ignoring it.

And also we are talking about a very rare scenario where someone would try to pass some bad data in skin parameter that usually happens when someone wants to do XSS attack, not in a day to day affair

So I would prefer to have constancy in code and have errors in log so admins would be aware about the attack

Reply:
Unpatched instance log is attached to initial bug: https://bugzilla.zimbra.com/attachment.cgi?id=68235. This used to happen seberal times a day  as bots scanned the Zimbra instances from different IPs.

Patched instance produces the following log entry in mailbox.log (not zmmailboxd.out!):

`2019-05-24 05:48:40,307 WARN  [qtp1647809929-67268:https:https://xx.xx.xx.xx/res/I18nMsg,AjxMsg,ZMsg,ZmMsg,AjxKeys,ZmKeys,ZdMsg,Ajx%20TemplateMsg.j
s.zgz?v=091214175450&skin=../../../../../../../../../opt/zimbra/conf/localconfig.xml%00] [] misc - Rejecting request containing null character in que
ry string`

so admins still have the alert for this type of attack.

And this is not an XSS attack, it is sort of local inclusion attack (just like XXE).

Reply:
@silentsakky so what's the final decision on this PR?

Reply:
ok

Reply:
Thanks! :)